test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -162,24 +163,31 @@ database_operation::insert_operation(thread_context *tc)
         auto &cc = ccv[counter];
         while (tc->transaction.active() && tc->running()) {
             /* Insert a key value pair. */
-            if (!tc->insert(cc.cursor, cc.coll.id, start_key + added_count)) {
-                committed = false;
-                break;
+            bool rollback_required = tc->insert(cc.cursor, cc.coll.id, start_key + added_count);
+            if (!rollback_required)
+                added_count++;
+
+            if (tc->transaction.can_commit_rollback()) {
```

Comment:
Can we rename `can_commit_rollback` to `can_commit_or_rollback` ? I got confused when re reading the code. And maybe add a comment in the header where the function is defined.

Reply:
I'll split them into two.

Reply:
Same is in `database_operation::read_operation`

Reply:
It checks it internally as well but I'll add it.